### PR TITLE
Improved doc on BufferGeometry.dispose()

### PR DESCRIPTION
**Pull request by erasta**
Related issue: #24607

Improved documentation on `BufferGeometry.dispose()` with a link to the separate page on objects disposing.
Hopefully this will better explain when `dispose()` is needed and when it can be avoided.
Example of a scenario that wasn't understood from the previous doc: when a geometry is not rendered, it doesn't have to be disposed.

### CODE REVIEW

File: docs/api/en/core/BufferGeometry.html [hunk 1]
```diff
@@ -222,7 +222,9 @@ <h3>[method:BufferAttribute deleteAttribute]( [param:String name] )</h3>
 		<h3>[method:undefined dispose]()</h3>
 		<p>
 		Disposes the object from memory. <br />
```
Comment (Mugen87) [inline docs/api/en/core/BufferGeometry.html:224]: I would remove this line. It's too unspecific.

File: docs/api/en/core/BufferGeometry.html [hunk 2]
```diff
@@ -222,7 +222,9 @@ <h3>[method:BufferAttribute deleteAttribute]( [param:String name] )</h3>
...
-		You need to call this when you want the BufferGeometry removed while the application is running.
+		When the geometry is rendered it allocates WebGL resources which are not implicitly freed.
+		You need to call this when you want the BufferGeometry removed while the application is running. <br />
+		For more details read: [link:https://threejs.org/docs/index.html#manual/en/introduction/How-to-dispose-of-objects How to dispose of objects]
```
Comment (Mugen87) [inline docs/api/en/core/BufferGeometry.html:227]: I would rephrase this section and write it more compact:

> When the geometry is rendered, it allocates WebGL resources which can only be disposed of by calling this method. For more details read: https://threejs.org/docs/index.html#manual/en/introduction/How-to-dispose-of-objects

Comment (erasta) [reply docs/api/en/core/BufferGeometry.html]: Agreed.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!
